app/views/RoomMembersView/index.tsx

```diff
@@ -420,251 +506,91 @@ class RoomMembersView extends React.Component<IRoomMembersViewProps, IRoomMember
-					room_name: getRoomTitle(room)
-				})
-			});
-		} catch (e) {
-			log(e);
-		}
-		this.fetchRoomMembersRoles();
-	};
-
-	handleLeader = async (selectedUser: TUserModel, isLeader: boolean) => {
-		try {
-			const { room } = this.state;
-			await Services.toggleRoomLeader({
-				roomId: room.rid,
-				t: room.t,
-				userId: selectedUser._id,
-				isLeader
-			});
-			const message = isLeader
-				? 'User__username__is_now_a_leader_of__room_name_'
-				: 'User__username__removed_from__room_name__leaders';
-			EventEmitter.emit(LISTENER, {
-				message: I18n.t(message, {
-					username: this.getUserDisplayName(selectedUser),
-					room_name: getRoomTitle(room)
-				})
-			});
-		} catch (e) {
-			log(e);
-		}
-		this.fetchRoomMembersRoles();
-	};
-
-	handleModerator = async (selectedUser: TUserModel, isModerator: boolean) => {
-		try {
-			const { room } = this.state;
-			await Services.toggleRoomModerator({
-				roomId: room.rid,
-				t: room.t,
-				userId: selectedUser._id,
-				isModerator
-			});
-			const message = isModerator
-				? 'User__username__is_now_a_moderator_of__room_name_'
-				: 'User__username__removed_from__room_name__moderators';
-			EventEmitter.emit(LISTENER, {
-				message: I18n.t(message, {
-					username: this.getUserDisplayName(selectedUser),
-					room_name: getRoomTitle(room)
-				})
-			});
-		} catch (e) {
-			log(e);
-		}
-		this.fetchRoomMembersRoles();
-	};
-
-	handleIgnore = async (selectedUser: TUserModel, ignore: boolean) => {
-		try {
-			const { room } = this.state;
-			await Services.ignoreUser({
-				rid: room.rid,
-				userId: selectedUser._id,
-				ignore
-			});
-			const message = I18n.t(ignore ? 'User_has_been_ignored' : 'User_has_been_unignored');
-			EventEmitter.emit(LISTENER, { message });
-		} catch (e) {
-			log(e);
-		}
-	};
-
-	handleRemoveUserFromRoom = async (selectedUser: TUserModel) => {
-		try {
-			const { room, members, membersFiltered } = this.state;
-			const userId = selectedUser._id;
-			// TODO: interface SubscriptionType on IRoom is wrong
-			await Services.removeUserFromRoom({ roomId: room.rid, t: room.t as RoomTypes, userId });
-			const message = I18n.t('User_has_been_removed_from_s', { s: getRoomTitle(room) });
-			EventEmitter.emit(LISTENER, { message });
-			this.setState({
-				members: members.filter(member => member._id !== userId),
-				membersFiltered: membersFiltered.filter(member => member._id !== userId)
-			});
-		} catch (e) {
-			log(e);
-		}
-	};
-
-	renderSearchBar = () => <SearchBox onChangeText={text => this.onSearchChangeText(text)} testID='room-members-view-search' />;
-
-	renderItem = ({ item }: { item: TUserModel }) => {
-		const { theme } = this.props;
-
-		return (
-			<UserItem
-				name={item.name as string}
-				username={item.username}
-				onPress={() => this.onPressUser(item)}
-				testID={`room-members-view-item-${item.username}`}
-				theme={theme}
+	const Member = React.memo(({ member }: { member: TUserModel }) => (
```

Comment:
We shouldn't add Nested Components.
Also, memo is `receiving` an observable obj from WatermelonDB and it's not memoizing the component.